src/ParametricCliqStateMachine.jl

```diff
@@ -0,0 +1,337 @@
+# STATUS messages [:initialized;:upsolved;:marginalized;:downsolved;:uprecycled]
```

Comment:
sorry, yeah we need to document the CSM better.  Guess we can start here JuliaRobotics/Caesar.jl#440

Reply:
Good idea, that will help. If this is more defined we can maybe use enumerated types?

Reply:
Symbols are like enumerated types (C++) or are you talking about hard types -- just want to avoid having too many abstract types floating around?

Reply:
Neither, https://docs.julialang.org/en/v1/base/base/#Base.Enums.@enum

Reply:
I would prefer `enum` over `Symbol` for ease of use and docstrings
```julia
julia> IIF.CliqStatus
Enum IncrementalInference.CliqStatus:
initialized = 0
upsolved = 1
marginalized = 2
downsolved = 3
uprecycled = 4

help?> IIF.CliqStatus
  primitive type CliqStatus <: Enum{Int32} 32

  Clique status message enumerated type
```